### PR TITLE
feat: limit trainee ltft endpoints to rollout PMs

### PR DESCRIPTION
**Pull request by ReubenRobertsHEE**
Checks on creating/updating a LTFT.

Note, no special status / message is returned from the API calls if they fail, just the generic 'Bad Request'. If we need this, should it be in a follow-up ticket, or added as an A/C?

TIS21-7305

### CODE REVIEW

**Review by Judge40 — CHANGES_REQUESTED** [review]
Considering the allowed programmes are going to be added to the token, we could get away with checking the linked programme ID against the token's list and avoid the service-to-service call.

File: src/integrationTest/java/uk/nhs/hee/tis/trainee/forms/TestJwtUtil.java [hunk 1]
```diff
@@ -70,15 +61,28 @@ public static String generateTokenForTisId(String traineeTisId) {
...
     return generateToken(payload);
   }
 
+  /**
+   * Generate a token with default values for the various attributes as the payload.
+   *
+   * @param traineeTisId The TIS ID to inject in to the payload.
+   * @return The generated token.
+   */
+  public static String generateTokenForTrainee(String traineeTisId) {
```
Comment (ReubenRobertsHEE) [inline src/integrationTest/java/uk/nhs/hee/tis/trainee/forms/TestJwtUtil.java:82]: This is a bit clunky, but replaces the 'generateTokenForTisId()' that various tests were using, to ensure that missing features don't either break tests or mask negative tests.

File: src/integrationTest/java/uk/nhs/hee/tis/trainee/forms/api/LtftResourceIntegrationTest.java [hunk 2]
```diff
@@ -654,10 +740,14 @@ void shouldIgnoreReadOnlyFieldsWhenUpdatingLtftFormForTrainee() throws Exception
             .build())
         .created(Instant.EPOCH)
         .lastModified(Instant.EPOCH)
+        .programmeMembership(LtftFormDto.ProgrammeMembershipDto.builder()
+            .id(PM_UUID)
+            .build())
         .build();
 
+    sleep(10); // Ensure lastModified is different from created
```
Comment (ReubenRobertsHEE) [inline src/integrationTest/java/uk/nhs/hee/tis/trainee/forms/api/LtftResourceIntegrationTest.java:748]: Blazing fast laptop, what can I say 😅

**Review by Judge40 — DISMISSED** [review]

**Comment by Judge40** [comment]
> > Considering the allowed programmes are going to be added to the token, we could get away with checking the linked programme ID against the token's list and avoid the service-to-service call.
> 
> I've refactored, but sadly the no. of lines has only bloated somewhat. It's the new need for programme membership data in the LTFT form that has swelled the tests, mostly.

I'll take that over inter-service API calls all day long 😄
